controllers/capabilities/openapi_product_reconciler.go

```diff
@@ -369,3 +375,112 @@ func (p *OpenAPIProductReconciler) desiredPrivateAPISecurity() *capabilitiesv1be
+	tmpHeaders := "headers"
+	authSpec.OIDC.CredentialsLoc = &tmpHeaders
+
+	if p.openapiCR.Spec.OIDC != nil {
+		authSpec.OIDC.IssuerType = p.openapiCR.Spec.OIDC.IssuerType
+		authSpec.OIDC.IssuerEndpoint = p.openapiCR.Spec.OIDC.IssuerEndpoint
+		authSpec.OIDC.IssuerEndpoint = p.getIssuerEndpoint(p.Client())
+		authSpec.OIDC.JwtClaimWithClientID = p.openapiCR.Spec.OIDC.JwtClaimWithClientID
+		authSpec.OIDC.JwtClaimWithClientIDType = p.openapiCR.Spec.OIDC.JwtClaimWithClientIDType
+
+		if p.openapiCR.Spec.OIDC.AuthenticationFlow != nil {
+			authSpec.OIDC.AuthenticationFlow.StandardFlowEnabled = p.openapiCR.Spec.OIDC.AuthenticationFlow.StandardFlowEnabled
+			authSpec.OIDC.AuthenticationFlow.ImplicitFlowEnabled = p.openapiCR.Spec.OIDC.AuthenticationFlow.ImplicitFlowEnabled
+			authSpec.OIDC.AuthenticationFlow.DirectAccessGrantsEnabled = p.openapiCR.Spec.OIDC.AuthenticationFlow.DirectAccessGrantsEnabled
+			authSpec.OIDC.AuthenticationFlow.ServiceAccountsEnabled = p.openapiCR.Spec.OIDC.AuthenticationFlow.ServiceAccountsEnabled
+		}
+	}
+}
+
+func (p *OpenAPIProductReconciler) parseOIDCCredentialsLoc(inField string) *string {
+	tmpQuery := "query"
+	tmpHeaders := "headers"
+	tmpAuthorisation := "authorisation"
+	switch inField {
+	case "query":
+		return &tmpQuery
+	case "header":
+		return &tmpHeaders
+	case "authorisation":
+		return &tmpAuthorisation
+	default:
+		return nil
+	}
+}
+
+func (p *OpenAPIProductReconciler) setOauth2AuthenticationParams(authSpec *capabilitiesv1beta1.AuthenticationSpec, secReq *helper.ExtendedSecurityRequirement) {
+	*authSpec.OIDC.CredentialsLoc = "header"
+
+	if p.openapiCR.Spec.OIDC != nil {
+		authSpec.OIDC.IssuerType = p.openapiCR.Spec.OIDC.IssuerType
+		authSpec.OIDC.IssuerEndpoint = p.openapiCR.Spec.OIDC.IssuerEndpoint
+		authSpec.OIDC.JwtClaimWithClientID = p.openapiCR.Spec.OIDC.JwtClaimWithClientID
+		authSpec.OIDC.JwtClaimWithClientIDType = p.openapiCR.Spec.OIDC.JwtClaimWithClientIDType
+
+		if secReq.Value.Flows.AuthorizationCode != nil {
+			authSpec.OIDC.AuthenticationFlow.StandardFlowEnabled = true
+		}
+		if secReq.Value.Flows.Implicit != nil {
+			authSpec.OIDC.AuthenticationFlow.ImplicitFlowEnabled = true
+		}
+		if secReq.Value.Flows.Password != nil {
+			authSpec.OIDC.AuthenticationFlow.DirectAccessGrantsEnabled = true
+		}
+		if secReq.Value.Flows.ClientCredentials != nil {
+			authSpec.OIDC.AuthenticationFlow.ServiceAccountsEnabled = true
+		}
+	}
+}
+
+func (p *OpenAPIProductReconciler) getIssuerEndpoint(cl k8sclient.Client) string {
+	// get IssuerEndpoint from Openapi CR and replace the stub value of "some-secret" with value
+	// of "secret" field of oidc-issuer-client-secret
+	var secret *string
+	crIssuerEndpoint := p.openapiCR.Spec.OIDC.IssuerEndpoint
+
+	oidcSecret, err := helper.GetSecret("oidc-issuer-client-secret", p.openapiCR.Namespace, cl)
```

Comment:
fixed name? What if there are two OpenAPI CRs with different clients?

Reply:
done. Thank you